presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -130,6 +135,24 @@ protected InternalResourceGroup(Optional<InternalResourceGroup> parent, String n
     public ResourceGroupInfo getInfo()
     {
         synchronized (root) {
+            if (subGroups.isEmpty()) {
```

Comment:
I don't think you need this `if`. You can just merge it with the else code path, by always doing `runningQueries.size() + descendantRunningQueries` and the like (one of them is going to be zero)